File: clientJava/src/main/java/at/rtr/rmbt/client/UuidStore.java
package at.rtr.rmbt.client;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

/** Persists the server-assigned client UUID to ~/.rmbt_client_uuid. */
final class UuidStore {

    private UuidStore() {}

    private static Path filePath() {
        String home = System.getenv("HOME");
        if (home == null) home = System.getenv("USERPROFILE");
        if (home == null) home = System.getProperty("user.home");
        return Path.of(home, ".rmbt_client_uuid");
    }

    /** Returns the stored UUID, or null if none exists yet. */
    static String load() {
        try {
            String s = Files.readString(filePath()).strip();
            return s.isEmpty() ? null : s;
        } catch (IOException e) {
            return null;
        }
    }

    /** Persist a server-assigned UUID. */
    static void save(String uuid) {
        Path path = filePath();
        try {
            Files.writeString(path, uuid + "\n");
            System.out.printf("Client UUID saved: %s%n  (%s)%n%n", uuid, path);
        } catch (IOException e) {
            System.err.println("Warning: could not save UUID to " + path + ": " + e.getMessage());
        }
    }
}

File: clientRust/src/uuid_store.rs
use std::fs;
use std::path::PathBuf;

fn uuid_file_path() -> Option<PathBuf> {
    let home = std::env::var_os("HOME")
        .or_else(|| std::env::var_os("USERPROFILE"))
        .map(PathBuf::from)?;
    Some(home.join(".rmbt_client_uuid"))
}

/// Load a persisted UUID from disk.  Returns None if no file exists yet.
pub fn load() -> Option<String> {
    let path = uuid_file_path()?;
    let s = fs::read_to_string(&path).ok()?;
    let uuid = s.trim().to_string();
    if uuid.is_empty() { None } else { Some(uuid) }
}

/// Persist a UUID received from the control server.
pub fn save(uuid: &str) {
    let path = match uuid_file_path() {
        Some(p) => p,
        None => {
            eprintln!("Warning: cannot determine home directory; UUID will not be persisted");
            return;
        }
    };
    match fs::write(&path, uuid) {
        Ok(_)  => println!("Client UUID saved: {uuid}\n  ({})", path.display()),
        Err(e) => eprintln!("Warning: could not save UUID to {}: {e}", path.display()),
    }
}
